tests/tests.h

```diff
@@ -187,6 +189,40 @@ namespace Testing
 
 
 
+// Get a uniformly distributed random value between min and max
+template<typename T=double>
+T random_value(const T &min, const T &max)
```

Comment:
are you using this for T!=double anywhere? What other types will work here? ints? unsigned ints?

Reply:
not yet in the submitted tests. I have this in some examples here (that we have not submitted yet) that select random indices using `unsigned int`.